src/cargo/util/toml.rs

```diff
@@ -737,7 +737,12 @@ impl TomlManifest {
                -> CargoResult<Vec<(PackageIdSpec, Dependency)>> {
         let mut replace = Vec::new();
         for (spec, replacement) in self.replace.iter().flat_map(|x| x) {
-            let spec = try!(PackageIdSpec::parse(spec));
+            let spec = match PackageIdSpec::parse(spec) {
+                Ok(spec) => spec,
+                Err(_)   => bail!("replacements must specify a \
+                                   valid semver version \
+                                   to replace, but `{}` does not", spec),
+            };
```

Comment:
Yeah it's true that it's not *too* helpful, but one day it may be! In general it's also just good hygiene to not throw away errors.